src/main/webapp/app/assessment/assessment-locks/assessment-locks.component.html

```diff
@@ -37,7 +37,7 @@ <h2>
                     <td>{{ submission.submissionDate | date: 'MMM d, y HH:mm:ss' }}</td>
                     <td>{{ submission.participation.submissions?.length }}</td>
                     <td>
-                        <span *ngIf="getLatestSubmissionResult(submission)!.score != null">{{ getLatestSubmissionResult(submission)!.score }}%</span>
+                        <span *ngIf="submission.latestResult!.score != null">{{ submission.latestResult!.score }}%</span>
```

Comment:
Please avoid the `!` operator. We never know if `latestResult` has a proper value.
Instead use the `?` operator so that it can fail.
Also avoid `null` and rather use `undefined` checks (I know this was here before, but we should change it)

Reply:
```suggestion
                        <span *ngIf="submission.latestResult?.score != undefined">{{ submission.latestResult.score }}%</span>
```